### ISSUE TITLE
`remark-lint-no-missing-blank-lines`

### ISSUE THREAD

**Issue by Janpot** [remarkjs/remark-lint#335]
### Initial checklist

- [x] I read the [support docs](https://github.com/remarkjs/.github/blob/main/support.md)
- [x] I read the [contributing guide](https://github.com/remarkjs/.github/blob/main/contributing.md)
- [x] I agree to follow the [code of conduct](https://github.com/remarkjs/.github/blob/main/code-of-conduct.md)
- [x] I searched [issues](https://github.com/search?q=user%3Aremarkjs&type=issues) and [discussions](https://github.com/orgs/remarkjs/discussions) and couldn’t find anything (or linked relevant results below)

### Affected package

`remark-lint-no-missing-blank-lines`

### Steps to reproduce

https://stackblitz.com/edit/stackblitz-starters-esr5apfz?file=index.mjs

Lint markdown:

```
> - item 1
> - item 2
>
> paragraph
```

### Actual behavior

```
4:3  Unexpected `0` blank lines between nodes, expected `1` or more blank lines, add `1` blank line
```

### Expected behavior

No issue

### Runtime

node 22.22.0

### Package manager

pnpm 8.15.6

### Operating system

Stackblitz

### Build and bundle tools

other (please specify in steps to reproduce)

**Comment by Amachua** [comment]
Hello teams, 

I have a similar error that is affecting the `remark-lint-list-item-spacing` rules.

A minimal example:

```md
> - Test
>   - Subtest
>   - Subtest
>
> - Test
>   - Subtest
>   - Subtest

Some additional test
```
 
Output:

```bash
4:2-5:3     warning Unexpected `0` blank lines between list items, expected `1` blank line, add `1` blank line list-item-spacing remark-lint
  [cause]:
    1:3-4:2 info    Spaced list item first defined here                                                        list-item-spacing remark-lint
```

Desired output:

No issue as this is only happening when having the line prefixed with ">".


Would it be possible to have this fixed within this issue or would you like me to open another issue?
